x-pack/legacy/plugins/ml/server/models/results_service/results_service.js

```diff
@@ -402,11 +402,111 @@ export function resultsServiceProvider(callWithRequest) {
     return definition;
   }
 
+  /**
+   * Gets the record of partition filed - values pairs.
+   * @param criteriaFields
+   * @param earliestMs
+   * @param latestMs
+   * @returns {Promise<*>}
+   */
+  async function getPartitionFieldsValues(criteriaFields, earliestMs, latestMs) {
```

Comment:
I'm not sure about the structure of the query+aggs here.
For example `partition_field_name` will return all field names, and `partition_field_values` will return all values.
Is the query done so we can be sure there will be only one `partition_field_name` returned? Otherwise the values could refer to different names.
To be more on the safe side, the aggregations could be done nested so the values agg is nested below the names agg. That way we'd have a definitive relationship between names+values defined in the response data.